src/sentry/integrations/slack/actions/notification.py

```diff
@@ -41,12 +42,14 @@ def after(self, event: GroupEvent, state: EventState) -> Generator[CallbackFutur
         channel = self.get_option("channel_id")
         tags = set(self.get_tags_list())
 
-        try:
-            integration = self.get_integration()
-        except Integration.DoesNotExist:
+        i = self.get_integration()
```

Comment:
Weird mypy issue -- the `Optionality`, although branched with the if, is captured into a closure and still looks Optional, thus breaking the uses below.  Splitting this into two variables and asserting the non optionality after the conditional fixes it.

Reply:
Aka no `final` in python

Reply:
Will using `assert isinstance(i, RpcInstallation)` help with narrowing after the `if not i` condition?